### PR TITLE
CORE-5663: test pipeline for combined worker

### PR DESCRIPTION
**Pull request by E1P3**

### CODE REVIEW

**Review by ronanbrowne — APPROVED** [review]
Approved from Jenkins file POV

File: libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/consumer/DBCordaConsumerImpl.kt [hunk 1]
```diff
@@ -232,6 +232,10 @@ internal class DBCordaConsumerImpl<K : Any, V : Any> constructor(
     internal fun getNextTopicPartition(): CordaTopicPartition? {
         updateTopicPartitions()
 
+        if (topicPartitions.isEmpty()) {
```
Comment (JamesHR3) [inline libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/consumer/DBCordaConsumerImpl.kt]: I don't think we should be seeing this diff if you're trying to merge into the release branch - have you synced up properly?

File: .ci/e2eTests/JenkinsfileCombinedWorker [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                sh './gradlew smoketest -PisCombinedWorker=true'
+    
+            }
+            // we need the DB credetals here also
+            
+        }
+    }
+    post {
+        always {
+	// remove Database see here https://github.com/corda/corda-shared-build-pipeline-steps/blob/5.0/vars/cordaPipeline.groovy#L487-L500
```
Comment (ronanbrowne) [inline .ci/e2eTests/JenkinsfileCombinedWorker]: Sorry @E1P3 actually few minor things prior to merge, please clean up redundant comments

File: libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/persistence/DBAccess.kt [hunk 3]
```diff
@@ -27,8 +27,9 @@ import javax.persistence.PersistenceException
 class DBAccess(
     private val entityManagerFactory: EntityManagerFactory,
 ) {
-
-    private val defaultNumPartitions = 1
+    // At the moment it's not easy to create partitions, so default value increased to 3 until tooling is available
+    // (There are multiple consumers using the same group for some topics and some stay idle if there is only 1 partition)
+    private val defaultNumPartitions = 3
```
Comment (JamesHR3) [inline libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/persistence/DBAccess.kt]: This is being committed under https://github.com/corda/corda-runtime-os/pull/1816 right?

File: .ci/e2eTests/JenkinsfileCombinedWorker [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                JAR_PATH = "${env.WORKSPACE}/applications/workers/release/combined-worker/build/bin/corda-combined-worker-5.0.0.0-SNAPSHOT.jar"
+                VM_PARAMETERS = "-Dco.paralleluniverse.fibers.verifyInstrumentation=true"
+                LOG4J_PARAMETERS = "-Dlog4j2.debug=-false -Dlog4j.configurationFile=log4j2-console.xml"
+                PROGRAM_PARAMETERS = "--instanceId=0 -mbus.busType=DATABASE -spassphrase=password -ssalt=salt -ddatabase.user=u${postgresDb} -ddatabase.pass=password -ddatabase.jdbc.url=jdbc:postgresql://${postgresHost}:${postgresPort}/${postgresDb}"
+                WORKING_DIRECTORY = "${env.WORKSPACE}"
+            }
+            steps {
+        
+                //JAVA_TOOL_OPTS env var to pass extra paramaters such as vm paramaters
+                //somehow ensure this runs in background and execute smoke tests , we may need some paramaters specific to combined worker
```
Comment (ronanbrowne) [inline .ci/e2eTests/JenkinsfileCombinedWorker]: remove these comments - these were just some suggestions i think when we started

**Review by JamesHR3 — COMMENTED** [review]
Looks like there's some extra code coming in that should already be in the release branch - can you double check that you're properly synced up with the release branch?

File: .ci/e2eTests/JenkinsfileCombinedWorker [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                                    sh 'psql --quiet --tuples-only -c \'select \''
+                                }
+                            }
+                            sh 'createdb -w "${DATABASE}"'
+                        }
+                    }
+                }
+            }
+        }
+    /// Build the Jar
```
Comment (ronanbrowne) [inline .ci/e2eTests/JenkinsfileCombinedWorker]: remove comment

File: .ci/e2eTests/JenkinsfileCombinedWorker [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+String postgresPort = '5432'
+@Field
+String postgresCredentialsId = 'e2e-postgresql-credentials'
+@Field
+String postgresDb = "test_${UUID.randomUUID().toString().replace("-", "")}"
+
+def checkConnection(){
+
+    def status = null
+
```
Comment (ronanbrowne) [inline .ci/e2eTests/JenkinsfileCombinedWorker]:
```suggestion
```

Spacing is a little odd in this method id tighten it up a bit

File: .ci/e2eTests/JenkinsfileCombinedWorker [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    sleep(time: 30, unit: "SECONDS") 
+
+    while (status != "200"){
+
```
Comment (ronanbrowne) [inline .ci/e2eTests/JenkinsfileCombinedWorker]:
```suggestion
```

As above

File: .ci/e2eTests/JenkinsfileCombinedWorker [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+        try{
+            status = sh(script: 'curl -s -o /dev/null -I -w "%{http_code}" http://localhost:7000/status', returnStdout: true)
+        }catch(error){
+            echo "can't connect, retrying..."
+        }
+
+        echo "${status}"
+          
+    }
+
```
Comment (ronanbrowne) [inline .ci/e2eTests/JenkinsfileCombinedWorker]:
```suggestion
```
as above

File: .ci/e2eTests/JenkinsfileCombinedWorker [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+                    nohup java ${LOG4J_PARAMETERS} -jar ${VM_PARAMETERS} ${JAR_PATH} ${PROGRAM_PARAMETERS} 2<&1 > workerLogs.txt &
+                    export PROCNO=$!
+                '''
+
+                checkConnection()    
+
+                sh './gradlew smoketest -PisCombinedWorker=true'
+    
+            }
+            // we need the DB credetals here also
```
Comment (ronanbrowne) [inline .ci/e2eTests/JenkinsfileCombinedWorker]: Redundant comment

**Review by ronanbrowne — CHANGES_REQUESTED** [review]
See comments

**Review by ronanbrowne — DISMISSED** [review]
All my comments on Jenkins file have been resolved thanks @E1P3 - leaving this in @mbrkic-r3   hands to address anything on kotlin side @JamesHR3 has called out

File: libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/consumer/DBCordaConsumerImpl.kt (hunk 1 above)
Comment (mbrkic-r3) [reply libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/consumer/DBCordaConsumerImpl.kt]: It hasn't been synced up

File: libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/persistence/DBAccess.kt (hunk 3 above)
Comment (mbrkic-r3) [reply libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/persistence/DBAccess.kt]: yes

**Review by ronanbrowne — DISMISSED** [review]
LGTM

File: libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/consumer/DBCordaConsumerImpl.kt (hunk 1 above)
Comment (mbrkic-r3) [reply libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/consumer/DBCordaConsumerImpl.kt]: it's rebased now

**Review by driessamyn — APPROVED** [review]
LGTM

**Review by JamesHR3 — APPROVED** [review]

**Pull request merged**